lib/rules/button-has-type.js

```diff
@@ -149,9 +150,11 @@ module.exports = {
         }
 
         const props = node.arguments[1].properties;
```

Comment:
is the type of `node` properly narrowed to a CallExpression node here? if not, let's do that as an annotation on the visitor function rather than needing to add all these runtime changes.

Reply:
I attempted to make corrections in 040d77c, but is my understanding correct?

Reply:
That's an acceptable narrowing :-) thanks